src/llm_needle_haystack_tester.py

```diff
@@ -210,20 +224,21 @@ def result_exists(self, context_length, depth_percent):
         Checks to see if a result has already been evaluated or not
         """
 
-        results_dir = 'results/'
-        if not os.path.exists(results_dir):
+        if not os.path.exists(self.results_dir):
+            return False
+        
+        filename = self.result_file_format.format(model_name=self.model_name,
+                                                  context_length=context_length,
+                                                  depth_percent=depth_percent)
+        file_path = os.path.join(self.results_dir, f'{filename}.json')
+        if not os.path.exists(file_path):
```

Comment:
We know the exact format of the filename, so instead of looping over every file lets see if the specific file exists, if it does then check the version number else return false.